File: Codepath/Week5-StacksAndQueues/java/StackUsingQueues.java
/*
 * Problem: Given a Queue data structure that supports standard operations like enqueue() and dequeue(),
 * implement a Stack data structure using only instances of Queue and queue operations allowed on the instances.
*/
static class Stack  
{
    private Queue<Integer> q1 = new LinkedList<>();
    private Queue<Integer> q2 = new LinkedList<>();
    private int top;

    public void push(int x) {
        q1.add(x);
        top = x;
    }

    public void pop() {
        while (q1.size() > 1) {
            top = q1.remove();
            q2.add(top);
        }
        q1.remove();
        Queue<Integer> temp = q1;
        q1 = q2;
        q2 = temp;
    }

    public int top()  {
       return top;
    }

    public boolean empty() {
        return q1.isEmpty();

File: Codepath/Week5-StacksAndQueues/python/StackUsingQueues.py
# Problem: Given a Queue data structure that supports standard operations like enqueue() and dequeue(),
# implement a Stack data structure using only instances of Queue and queue operations allowed on the instances.
import queue
class MyStack:
    q1 = queue.Queue()
    q2 = queue.Queue()
    top = -1

    def push(self, x):
        self.q1.put(x);
        self.top = x;

    def pop(self):
        while (self.q1.qsize() > 1):
            self.top = self.q1.get()
            self.q2.put(self.top)
        self.q1.get();
        temp = self.q1;
        self.q1 = self.q2;
        self.q2 = temp;

    def top(self):
        return top

    def empty(self):
        return self.q1.empty()
